js/okx.js

```diff
@@ -1650,10 +1650,11 @@ module.exports = class okx extends Exchange {
         const type = this.safeString (params, 'type', defaultType);
         params = this.omit (params, 'type');
         let method = 'publicGetMarket' + type;
+        const isHistoryCandles = (type === 'HistoryCandles');
```

Comment:
@sc0Vu shouldn't we also use this historic endpoint if since > 7 days?

Also, can point me to the docs pls? can't find these endpoints there

Reply:
@carlosmiei 

index https://www.okx.com/docs-v5/en/#rest-api-market-data-get-index-candlesticks-history
mark https://www.okx.com/docs-v5/en/#rest-api-market-data-get-mark-price-candlesticks-history https://www.okx.com/docs-v5/en/#rest-api-market-data-get-candlesticks-history

We probably can check whether the start time is 7 days before. eg,
`const isHistoryCandles = (type === 'HistoryCandles') || ( ( start - now ) > 7 * 86400 )`;

Reply:
> We probably can check whether the start time is 7 days before. eg,
> const isHistoryCandles = (type === 'HistoryCandles') || ( ( start - now ) > 7 * 86400 );

@sc0Vu exactly, if you can add it pls

Reply:
This might better:

```JS
if ((difference > limit * duration * 1000) || (difference > 7 * 86400000)) {
        defaultType = 'HistoryCandles';
}
```

Reply:
@sc0Vu actually, I made some tests and managed to retrieve data much older than 7 days so we can't rely on this 7 days logic (I was relying on the user's comment but apparently, it's not correct)

Example:
`n okx fetchOHLCV "BTC/USDT" "1h" 1665772914000 undefined '{"price":"mark"}'`

Reply:
@carlosmiei  They said it depends on the timeframe, the api only returns the latest 1440 entries (1m, the time would be 1 day). Let me test again.